src/main/java/com/zufar/onlinestore/cart/entity/ShoppingSession.java

```diff
@@ -0,0 +1,84 @@
+package com.zufar.onlinestore.cart.entity;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.UUID;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "shopping_session")
+public class ShoppingSession {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    private UUID id;
+
+    @Column(name = "user_id", nullable = false)
+    private UUID userId;
+
+    @OneToMany(mappedBy = "shoppingSession", cascade = CascadeType.ALL)
+    private Collection<ShoppingSessionItem> items;
+
+    @Column(name = "items_quantity", nullable = false)
+    private Integer itemsQuantity;
+
+    @Column(name = "products_quantity", nullable = false)
+    private Integer productsQuantity;
+
+    @Column(name = "created_at", nullable = false)
+    private LocalDateTime createdAt;
+
+    @Column(name = "closed_at", nullable = false)
+    private LocalDateTime closedAt;
+
+    @Override
+    public boolean equals(Object object) {
```

Comment:
for entity classes we should override "equals" only by id of entity
and the same for "hashCode"

Reply:
We don't use entities beyond the repository layer. 
Behind the layer of repositories we use model or dto objects

Reply:
Reason:

We don't use entities beyond the service layer.
Behind the layer of services we use model or dto objects

Equals and hashCode methods only need for Hibernate to manage entity properly and we never need to compare manually two entities by all fields at all.

Reply:
Reason:

Equals and hashCode methods are only needed for Hibernate to manage entity properly and we never need to compare manually two entities by all fields at all.

Entity is fully described by its id so we override equals and hashCode methods only by id

Reply:
Fixed